### PR TITLE
Fix getLudeonPath for non-Windows systems

### PR DESCRIPTION
**Pull request by Toby222**
- Replace `.unity3d` with `unity3d`
- Avoid NPE when creating path if `APPDATA` environment variable is not defined

### CODE REVIEW

**Comment by Garethp** [comment]
Hey Toby! Thanks for spotting the bug and contributing! Rather than splitting the list out into an `if/else` statement, do you mind just replacing the `.unity3d` with `unity3d` instead, and keeping it in the `listOf` with the `firstOrNull`? What `possiblePaths.firstOrNull { path: Path -> path.exists() }` does is that it fetches the first path in the list where the folder exists on the system, so if the `APPDATA` env var isn't defined it should already be skipping over that path.

In this project I'm using the pattern of defining a list of possible paths and then fetching the first one that exists in a few places since it makes it easier to support multiple systems without branching `if/else`'s. For example, if OSX has the directory at a different path I'd be able to just add the path to the `listOf` and it'd be picked out of that list without having to add another condition which checks if the OS is OSX or not.

Otherwise, looks good to me and I'd be happy to merge it in

File: src/rider/main/kotlin/run/QuickStartUtils.kt [hunk 1]
```diff
@@ -8,12 +8,14 @@ import kotlin.io.path.*
...
-            val possiblePaths = listOf(
-                Path(System.getenv("APPDATA"), "..", "LocalLow", "Ludeon Studios", "RimWorld by Ludeon Studios"),
-                Path(SystemUtils.getUserHome().absolutePath, ".config", ".unity3d", "Ludeon Studios", "RimWorld by Ludeon Studios"),
-            )
+            val osName = System.getProperty("os.name").lowercase(Locale.getDefault()) ?: return null
+            val systemPath =
+                if (osName.contains("win"))
+                    (Path(System.getenv("APPDATA"), "..", "LocalLow", "Ludeon Studios", "RimWorld by Ludeon Studios"))
+                else
+                    (Path(SystemUtils.getUserHome().absolutePath, ".config", "unity3d", "Ludeon Studios", "RimWorld by Ludeon Studios"))
```
Comment (feldoh) [inline src/rider/main/kotlin/run/QuickStartUtils.kt:16]:
```suggestion
            val possiblePaths = listOfNotNull(
                System.getenv("APPDATA")?.let{ Path(it, "..", "LocalLow", "Ludeon Studios", "RimWorld by Ludeon Studios") },
                Path(SystemUtils.getUserHome().absolutePath, ".config", "unity3d", "Ludeon Studios", "RimWorld by Ludeon Studios"),
            )
```

File: src/rider/main/kotlin/run/QuickStartUtils.kt [hunk 2]
```diff
@@ -8,12 +8,14 @@ import kotlin.io.path.*
...
 
-            return possiblePaths.firstOrNull { path: Path -> path.exists() }
+            return if (systemPath.exists()) systemPath else null
```
Comment (feldoh) [inline src/rider/main/kotlin/run/QuickStartUtils.kt:18]:
```suggestion
            return possiblePaths.firstOrNull { path: Path -> path.exists() }
```

**Comment by feldoh** [comment]
> Hey Toby! Thanks for spotting the bug and contributing! Rather than splitting the list out into an `if/else` statement, do you mind just replacing the `.unity3d` with `unity3d` instead, and keeping it in the `listOf` with the `firstOrNull`? What `possiblePaths.firstOrNull { path: Path -> path.exists() }` does is that it fetches the first path in the list where the folder exists on the system, so if the `APPDATA` env var isn't defined it should already be skipping over that path.

That doesn't actually work as is. The issue he was having was that the Path object construction with null in the first position NPEs because `System.getenv("APPDATA")` is null. It should probably be a let block instead using `listOfNotNull` I've provided a suggestion

**Pull request merged**